ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedWindowTableIQv2Test.java

```diff
@@ -0,0 +1,306 @@
+  @Mock
+  private KeyValueIterator<Windowed<GenericKey>, ValueAndTimestamp<GenericRow>> keyValueIterator;
+  @Captor
+  private ArgumentCaptor<QueryableStoreType<?>> storeTypeCaptor;
+
+  @Mock
+  private StateQueryResult queryResult;
+  @Mock
+  private QueryResult<WindowStoreIterator<ValueAndTimestamp<GenericRow>>> fetchResult;
+  @Mock
+  private QueryResult<KeyValueIterator<Windowed<GenericKey>, ValueAndTimestamp<GenericRow>>> keyValueResult;
+
+  private KsMaterializedWindowTableIQv2 table;
+
+  @Before
+  public void setUp() {
+    table = new KsMaterializedWindowTableIQv2(stateStore, WINDOW_SIZE);
+
+    when(stateStore.getStateStoreName()).thenReturn("rocksdb-store");
+    when(stateStore.getKafkaStreams()).thenReturn(kafkaStreams);
+    when(stateStore.schema()).thenReturn(SCHEMA);
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  @Test
+  public void shouldThrowNPEs() {
+    new NullPointerTester()
+        .setDefault(KsStateStore.class, stateStore)
+        .testConstructors(KsMaterializedWindowTable.class, Visibility.PACKAGE);
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFails() {
+    // Given:
+    when(stateStore.getKafkaStreams()).thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(A_KEY, PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to get value from materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFails_fetchAll() {
+    // Given:
+    when(stateStore.getKafkaStreams()).thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to scan materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfStoreFetchFails() {
+    // Given:
+    when(kafkaStreams.query(any()))
+        .thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(A_KEY, PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to get value from materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfStoreFetchFails_fetchAll() {
+    // Given:
+    when(kafkaStreams.query(any()))
+      .thenThrow(new MaterializationTimeOutException("Boom"));
```

Comment:
I think several of these tests are copied and modified from the IQv1 test (which is good), but I don't think they've been modified in a way that makes sense all the time. For example, this test is identical to the previous one.

Reply:
mmh, I can't find any duplicates